docs/temporal-explained/workflow-determinism.md

```diff
@@ -0,0 +1,131 @@
+---
+id: workflow-determinism
+title: Workflow Determinism
+sidebar_label: Workflow Determinism
+description: Why workflows must be deterministic, and how you can ensure they are.
+---
+
+# Workflow Determinism
+
+Temporal Workflows must be deterministic. Stated using our terminology:
+
+> For a given [Workflow Type](/docs/concepts/what-is-a-workflow-type), its [Workflow Definition](/docs/concepts/what-is-a-workflow-definition)
+> (implementation) must produce the same sequence of [Commands](/docs/concepts/what-is-a-command) given the same [History](/docs/concepts/what-is-an-event-history)
+
+Less formally, this means a Workflow must always do the same thing given the same inputs.
```

Comment:
Here `inputs` is meant generically, inclusive of workflow arguments and things like signals. In that context, the statement is definitely true, but I'll clarify that a bit more here.

Reply:
> this is not a correct statement.

The statement is correct in the abstract and I think a good one. Technically you can do different things with the same input and still be deterministic by Temporal standards. There are lots of caveats and details to the statement, but the "Do the same thing for the same input" is reasonable I think.

Reply:
> The statement is correct in the abstract
> There are lots of caveats and details to the statement

Ya, which is why we are not going to accept it.
Those type of "caveats" drive support questions that cause us to have to work super hard to reframe questions.
We want to avoid that - we want precision - we want accuracy.